Sources/NIO/BSDSocketAPI.swift

```diff
@@ -41,15 +54,107 @@ import let WinSDK.SO_RCVTIMEO
 import let WinSDK.SO_REUSEADDR
 import let WinSDK.SO_REUSE_UNICASTPORT
 
+import let WinSDK.SOL_SOCKET
+
 import let WinSDK.SOCK_DGRAM
 import let WinSDK.SOCK_STREAM
-#elseif os(iOS) || os(macOS) || os(tvOS) || os(watchOS)
-import Darwin
+
+import let WinSDK.SOCKET_ERROR
+
+import func WinSDK.WSAGetLastError
+
+import struct WinSDK.socklen_t
+import struct WinSDK.SOCKADDR
+
+internal typealias sockaddr = WinSDK.SOCKADDR
 #else
+@_exported
 import Glibc
 #endif
 
+#if !os(Windows)
+let cAccept: @convention(c) (CInt, UnsafeMutablePointer<sockaddr>?, UnsafeMutablePointer<socklen_t>?) -> CInt = accept
+let cBind: @convention(c) (CInt, UnsafePointer<sockaddr>?, socklen_t) -> CInt = bind
+let cClose: @convention(c) (CInt) -> CInt = close
+let cConnect: @convention(c) (CInt, UnsafePointer<sockaddr>?, socklen_t) -> CInt = connect
+let cGetpeername: @convention(c) (CInt, UnsafeMutablePointer<sockaddr>?, UnsafeMutablePointer<socklen_t>?) -> CInt = getpeername
+let cGetsockname: @convention(c) (CInt, UnsafeMutablePointer<sockaddr>?, UnsafeMutablePointer<socklen_t>?) -> CInt = getsockname
+let cGetsockopt: @convention(c) (CInt, CInt, CInt, UnsafeMutableRawPointer?, UnsafeMutablePointer<socklen_t>?) -> CInt = getsockopt
+let cListen: @convention(c) (CInt, CInt) -> CInt = listen
+let cRead: @convention(c) (CInt, UnsafeMutableRawPointer?, size_t, CInt) -> ssize_t = recv
+let cRecvfrom: @convention(c) (CInt, UnsafeMutableRawPointer?, size_t, CInt, UnsafeMutablePointer<sockaddr>?, UnsafeMutablePointer<socklen_t>?) -> ssize_t = recvfrom
+let cSend: @convention(c) (CInt, UnsafeRawPointer?, size_t, CInt) -> ssize_t = send
+let cSetsockopt: @convention(c) (CInt, CInt, CInt, UnsafeRawPointer?, socklen_t) -> CInt = setsockopt
+let cSendto: @convention(c) (CInt, UnsafeRawPointer?, size_t, CInt, UnsafePointer<sockaddr>?, socklen_t) -> ssize_t = sendto
+let cShutdown: @convention(c) (CInt, CInt) -> CInt = shutdown
+let cSocket: @convention(c) (CInt, CInt, CInt) -> CInt = socket
+
+let cPread: @convention(c) (CInt, UnsafeMutableRawPointer?, size_t, off_t) -> ssize_t = pread
+let cPwrite: @convention(c) (CInt, UnsafeRawPointer?, size_t, off_t) -> ssize_t = pwrite
+
+let cPoll: @convention(c) (UnsafeMutablePointer<pollfd>?, nfds_t, CInt) -> CInt = poll
+
+let cInet_ntop: @convention(c) (CInt, UnsafeRawPointer?, UnsafeMutablePointer<CChar>?, socklen_t) -> UnsafePointer<CChar>? = inet_ntop
+let cInet_pton: @convention(c) (CInt, UnsafePointer<CChar>?, UnsafeMutableRawPointer?) -> CInt = inet_pton
```

Comment:
I'm pretty sure these all used to be `sysX`: why the change?

Reply:
I generally am not fond of the prefixed aliases, but @weissi is adamant about them.  So, this works as a decent enough compromise I think - its a short prefix and indicates that they are from the C world.  Id be okay with `pfn` as well as they are closer to pointers to functions if you like.

Reply:
I was proposing keeping them as `sys`.

Reply:
Right, I'm suggesting that `c` is nicer than `sys` as a prefix - although, my personal preference would be to duplicate the code and use the module qualified name.

Reply:
My primary concern is to twofold:
- keep the diff as small as possible
- no duplication of code. Darwin & Linux match almost perfectly so we don't need to duplicate all code

Reply:
Why can't they also stay in `System.swift`? I always really liked to have just one place where I can find all the system calls that NIO makes.

Reply:
I think that having the BSD sockets API separated from the C library surface helps immensely in understanding the codebase (its read more than written), it helps discoverability, it helps porting, etc.  System.swift was immensely large, and it is going to have to grow to at least 3x its current size.